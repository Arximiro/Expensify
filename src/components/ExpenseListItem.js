import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem = ({id, description, note, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>Amount - {amount}</p>
        <p>Created At - {createdAt}</p>
        {note && <p>Note - {note}</p>}
    </div>
);

export default ExpenseListItem;

// Using connect in this manner will give us access to dispatch. Since we don't need access to state here we can just use connect
// to get access to what we need to which is the ability to dispatch.
