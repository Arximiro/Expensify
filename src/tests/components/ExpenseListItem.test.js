import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/testExpenses';

// --- Jest Notes ---
// ExpenseListItem is rendered passing in an expense from fixtures, spreading out the expense to get the needed properties.

test('Should render ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[1]} />)
    expect(wrapper).toMatchSnapshot();
});
