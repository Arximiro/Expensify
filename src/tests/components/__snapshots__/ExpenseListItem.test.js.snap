// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Car
    </h3>
  </Link>
  <p>
    Amount - 
    50800
  </p>
  <p>
    Created At - 
    86400000
  </p>
  <p>
    Note - 
    Civic
  </p>
</div>
`;
